src/iidm/converter/xml/AbstractConnectableXml.hxx

```diff
@@ -115,9 +155,38 @@ void AbstractConnectableXml<Added, Adder, Parent>::writeBus(const stdcxx::CRefer
     }
 }
 
+template <typename Added, typename Adder, typename Parent>
+void AbstractConnectableXml<Added, Adder, Parent>::writeActivePowerLimits(const ActivePowerLimits& limits, powsybl::xml::XmlStreamWriter& writer, const IidmXmlVersion& version, const stdcxx::optional<int>& index) {
+    writeLoadingLimits(limits, writer, version.getPrefix(), version, ACTIVE_POWER_LIMITS, index);
+}
+
+template <typename Added, typename Adder, typename Parent>
+void AbstractConnectableXml<Added, Adder, Parent>::writeApparentPowerLimits(const ApparentPowerLimits& limits, powsybl::xml::XmlStreamWriter& writer, const IidmXmlVersion& version, const stdcxx::optional<int>& index) {
+    writeLoadingLimits(limits, writer, version.getPrefix(), version, APPARENT_POWER_LIMITS, index);
+}
+
 template <typename Added, typename Adder, typename Parent>
 void AbstractConnectableXml<Added, Adder, Parent>::writeCurrentLimits(const CurrentLimits& limits, powsybl::xml::XmlStreamWriter& writer, const IidmXmlVersion& version, const stdcxx::optional<int>& index) {
-    CurrentLimitsXml::writeCurrentLimits(limits, writer, version.getPrefix(), version, index);
+    writeLoadingLimits(limits, writer, version.getPrefix(), version, CURRENT_LIMITS, index);
+}
+
+template <typename Added, typename Adder, typename Parent>
+template <typename Limits>
+void AbstractConnectableXml<Added, Adder, Parent>::writeLoadingLimits(const Limits& limits, powsybl::xml::XmlStreamWriter& writer, const std::string& nsPrefix, const IidmXmlVersion& version, const std::string& type, const stdcxx::optional<int>& index) {
+    if (!std::isnan(limits.getPermanentLimit()) || !boost::empty(limits.getTemporaryLimits())) {
+        writer.writeStartElement(nsPrefix, toString(type.c_str(), index));
+        writer.writeAttribute(PERMANENT_LIMIT, limits.getPermanentLimit());
+
+        for (const auto& tl : limits.getTemporaryLimits()) {
```

Comment:
In Java implementation, temporaryLimits seems to be sorted. Is it already the case with the C++ map?

Reply:
Yes they are because `LoadingLimits::m_temporaryLimits` is a `std::map<unsigned long, TemporaryLimit, std::greater<unsigned long> >`